home/text.asm

```diff
@@ -115,6 +115,9 @@ NullChar::
 	ld b, h
 	ld c, l
 	pop hl
+	; This error message seems to be a development leftover.
+	; Removing the instruction below will cause the game to
+	; ignore <NULL> and print the text script normally.	
```

Comment:
Aka "Changing the implementation of printing `<NULL>` will change what `<NULL>` does." So? What does it seem to be a leftover of? I'm guessing this is a failsafe so that a `<NULL>` in the middle of a text string will be printed as an error message, right? If so, it's an error handler, not a leftover, and choosing to ignore nulls is the sort of customization people can make on their own (but why would you?), not a fact about the game that actually exists.

Reply:
> I'm guessing this is a failsafe so that a `<NULL>` in the middle of a text string will be printed as an error message, right?

The placement of `<NULL>` can be anywhere in the text script and what the routine does is discard the character. The error message returns the value of `hSpriteIndexOrTextID`, which is completely obscure and cryptic to the player.

Kindly note that [this never happens in regular gameplay](https://glitchcity.wiki/Error_codes_(Generations_I_and_II)#00_character_error_codes) and that they [didn't even bother translating the message later on](https://github.com/pret/pokegold/blob/d36f220b902ea210a3de86e1b7c0c7f938943121/home/text.asm#L502-L505).

Reply:
> The 00 character error codes (Japanese: エラー) appear in the form of "(X) ERROR" in Pokémon Red and Blue or "(X) error" in Pokémon Yellow when a 00 character appears in the middle of a text box. [...]
 The value of the error code is the index number of the text box. These error codes can commonly found when memory hacking, but they can also be found with the Trainer escape glitch [...]

Okay. In that case, something like:
```
; A "<NULL>" character in a printed string
; prints an error message with the index number
; of its textbox.
; This is a debugging leftover.
```

IMO there's no need to discuss what happens when arbitrary instructions are removed or changed. That's something for the wiki, although here it's still not necessary since nobody wants to allow a quiet no-effect `"<NULL>"` in their glitchy textboxes anyway.

(If a person really does want to change what the `"<NULL>"` character does, they can/should read over how the whole text.asm printing system works, and make the appropriate edits to do what they want. If the labels and comments so far don't seem good enough to make that feasible, a separate PR should fix that. Like these: <https://github.com/Rangi42/polishedcrystal/blob/9bit/home/text.asm#L171>)

Reply:
How about this?
```suggestion
	; A "<NULL>" character in a printed string
	; returns an error message with the current value
	; of hSpriteIndexOrTextID in decimal format.
	; This is a debugging leftover.
```

Reply:
"prints" or "displays" not "returns", but yes.